tests/components/jewish_calendar/test_service.py

```diff
@@ -12,44 +12,94 @@
 
 
 @pytest.mark.parametrize(
-    ("test_date", "nusach", "language", "expected"),
+    ("test_time", "service_data", "expected"),
     [
-        pytest.param(dt.date(2025, 3, 20), "sfarad", "he", "", id="no_blessing"),
         pytest.param(
-            dt.date(2025, 5, 20),
-            "ashkenaz",
-            "he",
+            dt.datetime(2025, 3, 20, 21, 0),
+            {
+                "date": dt.date(2025, 3, 20),
+                "nusach": "sfarad",
+                "language": "he",
+                "is_after_sunset": False,
+            },
+            "",
+            id="no_blessing",
+        ),
+        pytest.param(
+            dt.datetime(2025, 3, 20, 21, 0),
+            {
+                "date": dt.date(2025, 5, 20),
+                "nusach": "ashkenaz",
+                "language": "he",
+                "is_after_sunset": False,
+            },
             "היום שבעה ושלושים יום שהם חמישה שבועות ושני ימים בעומר",
             id="ahskenaz-hebrew",
         ),
         pytest.param(
-            dt.date(2025, 5, 20),
-            "sfarad",
-            "en",
+            dt.datetime(2025, 3, 20, 21, 0),
+            {
+                "date": dt.date(2025, 5, 20),
+                "nusach": "sfarad",
+                "language": "en",
+                "is_after_sunset": True,
+            },
+            "Today is the thirty-eighth day, which are five weeks and three days of the Omer",
+            id="sefarad-english-after-sunset",
+        ),
+        pytest.param(
+            dt.datetime(2025, 3, 20, 21, 0),
+            {
+                "date": dt.date(2025, 5, 20),
+                "nusach": "sfarad",
+                "language": "en",
+                "is_after_sunset": False,
+            },
             "Today is the thirty-seventh day, which are five weeks and two days of the Omer",
-            id="sefarad-english",
+            id="sefarad-english-before-sunset",
+        ),
+        pytest.param(
+            dt.datetime(2025, 5, 20, 21, 0),
+            {"nusach": "sfarad", "language": "en"},
+            "Today is the thirty-eighth day, which are five weeks and three days of the Omer",
+            id="sefarad-english-after-sunset-without-date",
+        ),
+        pytest.param(
+            dt.datetime(2025, 5, 20, 6, 0),
+            {
+                "nusach": "sfarad",
+            },
+            "היום שבעה ושלושים יום שהם חמישה שבועות ושני ימים לעומר",
+            id="sefarad-english-before-sunset-without-date",
         ),
     ],
+    indirect=["test_time"],
 )
```

Comment:
Add:
```suggestion
)
@pytest.mark.usefixture("setup_at_time")
```

to remove the initial setup, as well as the need for using freezegun